pkg/apis/nfs.rook.io/v1alpha1/webhook.go

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"strings"
+
+	"github.com/coreos/pkg/capnslog"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/kustomize/kyaml/sets"
+)
+
+var (
+	webhookName = "nfs-webhook"
+	logger      = capnslog.NewPackageLogger("github.com/rook/rook", webhookName)
+)
+
+var _ webhook.Defaulter = &NFSServer{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *NFSServer) Default() {
+	logger.Info("default", "name", r.Name)
+	logger.Warning("defaulting is not supported yet")
+}
+
+var _ webhook.Validator = &NFSServer{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *NFSServer) ValidateCreate() error {
+	logger.Info("validate create", "name", r.Name)
+
+	if err := r.ValidateSpec(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
```

Comment:
`ValidateUpdate` will also register a webhook?
this comment is similar to the comment for `ValidateCreate` in line 44.

Reply:
updated, thanks